src/window.js

```diff
@@ -57,6 +57,21 @@ export default function Window({ application, session }) {
   }
   window.set_application(application);
 
+  function updateTitle () {
+    window.set_title (`Workbench — ${session.file.get_basename ()}`);
+  }
+
+  // Monitor the session directory and update the window title if it is renamed
+  // FIXME: The monitor does not seem to listen for rename events
+  let monitor = session.file.monitor (Gio.FileMonitorFlags.WATCH_MOVES, null);
```

Comment:
You need to hold a reference to this somewhere, or it will just be collected and the signal disconnected with it.

Reply:
Let's remove the monitor for now.

Currently, if the project is moved or renamed while being open, it'll break anyway.

I have created a follow-up to address the problem more globally https://github.com/sonnyp/Workbench/issues/468